var/spack/repos/builtin/packages/chapel/package.py

```diff
@@ -3,24 +3,813 @@
+        description="Host architecture of the build machine",
+        values=("x86_64", "aarch64", "arm64", "unset"),
+        default="unset",
+        multi=False,
+    )
+
+    variant(
+        "host_jemalloc",
+        values=("bundled", "none", "spack", "unset"),
+        default="unset",
+        multi=False,
+        description="Selects between no jemalloc, bundled jemalloc, or spack supplied jemalloc",
+    )
+
+    variant(
+        "host_mem",
+        values=("cstdlib", "jemalloc"),
+        default="jemalloc",
+        description="Memory management layer for the chpl compiler",
+        multi=False,
+    )
+
+    variant(
+        "host_platform",
+        description="Host platform",
+        default="unset",
+        values=platform_opts,
+        multi=False,
+    )
+
+    variant(
+        "hwloc",
+        description="Build with hwloc support",
+        default="bundled",
+        values=(
+            "bundled",
+            "none",
+            # CHPL_HWLOC=system existed back to at least 2017,
+            # but it was buggy and unsupported until version 2.1
+            conditional("spack", when="@2.1:"),
+        ),
+        multi=False,
+    )
+
+    variant(
+        "launcher",
+        values=launcher_names,
+        default="unset",
+        description="Launcher to use for running Chapel programs",
+        multi=False,
+    )
+
+    variant(
+        "lib_pic",
+        values=("none", "pic"),
+        default="none",
+        description="Build position-independent code suitable for shared libraries",
+    )
+
+    variant(
+        "libfabric",
+        default="unset",
+        description="When building with ofi support, specify libfabric option",
+        values=("bundled", "spack", "unset"),
+        multi=False,
+    )
+
+    variant(
+        "llvm",
+        default="spack",
+        description="LLVM backend type. The 'spack' value can use an external "
+        "source of LLVM or let spack build a version if no LLVM installs were "
+        "previously detected by 'spack external find'",
+        values=("bundled", "none", "spack"),
+    )
+
+    variant(
+        "re2",
+        description="Build with re2 support",
+        default="bundled",
+        values=("bundled", "none"),
+        multi=False,
+    )
+
+    variant(
+        "target_arch",
+        description="Target architecture for cross compilation",
+        default="unset",
+        values=("x86_64", "aarch64", "arm64", "unset"),
+        multi=False,
+    )
+
+    variant(
+        "target_cpu",
+        values=cpu_options,
+        description="Indicate that the target executable should be specialized "
+        "to the given architecture when using --specialize (and --fast).",
+        default="unset",
+        multi=False,
+    )
+
+    variant(
+        "target_platform",
+        description="Target platform for cross compilation",
+        default="unset",
+        values=platform_opts,
+        multi=False,
+    )
```

Comment:
(minor) I don't see this variant being used anywhere like (unlike `host_platform`, for example), likewise for `target_cpu`/`target_arch`/`timers`/`launcher`.

In https://github.com/spack/spack/pull/42197/files#r1631482021 you mention:

> this is another value that is just used inside the chapel build scripts, but it is important to specify when installing because of the heterogenous nature of some machines

what I mean is that while you unset the `"CHPL_TARGET_ARCH"` env variable, I do not see anywhere in this package that reads the value of this variant, so it cannot have an impact on the build.

Reply:
I think what I didn't explain very well was that all of the keys are defined in `chpl_env_vars` and the processing around line 561 evaluates each variant and sets the value in the environment if the variant was given anything other than 'unset' so that Chapel's build scripts can detect them and that will affect the build.